MPChartLib/src/com/github/mikephil/charting/charts/BarLineChartBase.java

```diff
@@ -527,14 +527,18 @@ public void calculateOffsets() {
     /**
      * calculates the modulus for x-labels and grid
      */
+    float[] valuesForCalcModulus = new float[9];
     protected void calcModulus() {
 
         if (mXAxis == null || !mXAxis.isEnabled())
             return;
 
         if (!mXAxis.isAxisModulusCustom()) {
 
-            float[] values = new float[9];
+            float[] values = valuesForCalcModulus;
+            for(int i = 0 ; i < values.length ; i++){
```

Comment:
This and all other similar items are addressed in my local branch based on the remove_xindeces branch.  remove_xindeces will soon be merged into master, and it introduces many changes that will make this pull redundant (eg: changes to the Axis renderers, completely removing Realm from the repo, etc).